Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BaseShapeComponentMode.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+#include <AzToolsFramework/ComponentModes/BaseShapeViewportEdit.h>
+#include <AzToolsFramework/ComponentModes/ShapeComponentModeBus.h>
+#include <AzToolsFramework/ComponentMode/EditorBaseComponentMode.h>
+
+namespace AzToolsFramework
+{
+    void InstallBaseShapeViewportEditFunctions(
+        BaseShapeViewportEdit* baseShapeViewportEdit, const AZ::EntityComponentIdPair& entityComponentIdPair);
+
+    class BaseShapeComponentMode
```

Comment:
suggestion: An API comment for this type would be super useful

Reply:
Added